vertical-pod-autoscaler/pkg/recommender/routines/recommender_routines_test.go

```diff
@@ -0,0 +1,267 @@
+package routines
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	vpa_types "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/recommender/logic"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/recommender/model"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/recommender/util"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/utils/test"
+	"k8s.io/klog"
+)
+
+// getOOMParam defines a function return type to handle tests
+// for both OOMBumpUpRatio and OOMMinBUmp
+type getOOMParam func(float64) float64
+
+var (
+	testPodID1       = model.PodID{Namespace: "namespace-1", PodName: "pod-1"}
+	testLabels       = map[string]string{"label-1": "value-1"}
+	timeLayout       = "2006-01-02 15:04:05"
+	testTimestamp, _ = time.Parse(timeLayout, "2021-08-11 14:53:05")
+	testRequest      = model.Resources{
+		model.ResourceCPU:    model.CPUAmountFromCores(0.5),
+		model.ResourceMemory: model.MemoryAmountFromBytes(100),
+	}
+	testLimit = model.Resources{ // Setting limits to 3x of testRequest
+		model.ResourceCPU:    model.CPUAmountFromCores(1.0),
+		model.ResourceMemory: model.MemoryAmountFromBytes(300),
+	}
+	testUpdateModeAuto = vpa_types.UpdateModeAuto
+	testSelectorStr    = "label-1 = value-1"
+	testVpaID          = model.VpaID{
+		Namespace: testPodID1.Namespace,
+		VpaName:   "testVPA",
+	}
+)
+
+func addTestMemorySample(cluster *model.ClusterState, container model.ContainerID, memoryBytes float64, timeStamp time.Time) error {
+	sample := model.ContainerUsageSampleWithKey{
+		Container: container,
+		ContainerUsageSample: model.ContainerUsageSample{
+			MeasureStart: timeStamp,
+			Usage:        model.MemoryAmountFromBytes(memoryBytes),
+			Request:      testRequest[model.ResourceMemory],
+			Resource:     model.ResourceMemory,
+		},
+	}
+	return cluster.AddSample(&sample)
+}
+
+func addTestCPUHistogram(samples []float64, weight float64, testTimeStamp time.Time, options util.HistogramOptions) util.Histogram {
+	cpuHistogram := util.NewHistogram(options)
+	for _, sample := range samples {
+		cpuHistogram.AddSample(sample, weight, testTimestamp)
+	}
+
+	return cpuHistogram
+}
+
+func addTestMemoryHistogram(samples []float64, weight float64, timeStamp time.Time, options util.HistogramOptions) util.Histogram {
+	memoryHistogram := util.NewHistogram(options)
+	for _, sample := range samples {
+		memoryHistogram.AddSample(sample, weight, testTimestamp)
+	}
+
+	return memoryHistogram
+}
```

Comment:
Same comments as for `addTestCPUHistogram`.

Reply:
ACK.